streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/ProcessorNodeMetrics.java

```diff
@@ -0,0 +1,424 @@
+        if (streamsMetrics.version() == Version.FROM_0100_TO_24) {
+            return throughputAndLatencySensorWithParent(
+                threadId,
+                taskId,
+                processorNodeId,
+                PROCESS,
+                PROCESS_RATE_DESCRIPTION,
+                PROCESS_TOTAL_DESCRIPTION,
+                PROCESS_AVG_LATENCY_DESCRIPTION,
+                PROCESS_MAX_LATENCY_DESCRIPTION,
+                RecordingLevel.DEBUG,
+                streamsMetrics
+            );
+        }
+        return emptySensor(threadId, taskId, processorNodeId, PROCESS, RecordingLevel.DEBUG, streamsMetrics);
+    }
+
+    public static Sensor processAtSourceSensor(final String threadId,
+                                               final String taskId,
+                                               final String processorNodeId,
+                                               final StreamsMetricsImpl streamsMetrics) {
+        final Sensor parentSensor = streamsMetrics.taskLevelSensor(threadId, taskId, PROCESS, RecordingLevel.DEBUG);
+        addInvocationRateAndCountToSensor(
+            parentSensor,
+            TASK_LEVEL_GROUP,
+            streamsMetrics.nodeLevelTagMap(threadId, taskId, ROLLUP_VALUE),
+            PROCESS,
+            PROCESS_RATE_DESCRIPTION,
+            PROCESS_TOTAL_DESCRIPTION
+        );
+        return throughputSensor(
+            threadId,
+            taskId,
+            processorNodeId,
+            PROCESS,
+            PROCESS_RATE_DESCRIPTION,
+            PROCESS_TOTAL_DESCRIPTION,
+            RecordingLevel.DEBUG,
+            streamsMetrics,
+            parentSensor
+        );
+    }
+
+    public static Sensor punctuateSensor(final String threadId,
+                                         final String taskId,
+                                         final String processorNodeId,
+                                         final StreamsMetricsImpl streamsMetrics) {
+        if (streamsMetrics.version() == Version.FROM_0100_TO_24) {
+            return throughputAndLatencySensorWithParent(
+                threadId,
+                taskId,
+                processorNodeId,
+                PUNCTUATE,
+                PUNCTUATE_RATE_DESCRIPTION,
+                PUNCTUATE_TOTAL_DESCRIPTION,
+                PUNCTUATE_AVG_LATENCY_DESCRIPTION,
+                PUNCTUATE_MAX_LATENCY_DESCRIPTION,
+                RecordingLevel.DEBUG,
+                streamsMetrics
+            );
+        }
+        return emptySensor(threadId, taskId, processorNodeId, PUNCTUATE, RecordingLevel.DEBUG, streamsMetrics);
+    }
+
+    public static Sensor createSensor(final String threadId,
+                                      final String taskId,
+                                      final String processorNodeId,
+                                      final StreamsMetricsImpl streamsMetrics) {
+        if (streamsMetrics.version() == Version.FROM_0100_TO_24) {
```

Comment:
Not a comment: I thought in old version, the create / destroy are at the task-level right, but seems they are on processor-node level? If that's indeed the case could you update the KIP wiki on the removed section as well?